pinot-spi/src/main/java/org/apache/pinot/core/accounting/WorkloadBudgetManager.java

```diff
@@ -47,10 +47,36 @@ public WorkloadBudgetManager(PinotConfiguration config) {
     }
     _enforcementWindowMs = config.getProperty(CommonConstants.Accounting.CONFIG_OF_WORKLOAD_ENFORCEMENT_WINDOW_MS,
         CommonConstants.Accounting.DEFAULT_WORKLOAD_ENFORCEMENT_WINDOW_MS);
+    initSecondaryWorkloadBudget(config);
```

Comment:
As discussed in earlier comments, we should keep this optional. Not all users will be migrating from BinaryWorkloadScheduler. If DEFAULT_SECONDARY_WORKLOAD_NAME is not populated, maybe we can skip initializing?